src/NexusMods.App.UI/Pages/ModLibrary/FileOrigins/FileOriginsPageViewModel.cs

```diff
@@ -1,73 +1,169 @@
+                        {
+                            LoadoutId = loadoutId,
+                        }
+                    };
 
-        var entriesObservable = downloadAnalysisRepository.Observable
+                    var behavior = workspaceController.GetOpenPageBehavior(pageData, info, IdBundle);
+                    workspaceController.OpenPage(workspaceController.ActiveWorkspace!.Id, pageData, behavior);
+                }
+            );
+            
+            var entriesObservable = dlAnalysisRepo.Observable
                 .ToObservableChangeSet()
                 .Filter(model => FilterDownloadAnalysisModel(model, game.Domain))
                 .OnUI()
-                .Transform(fileOrigin => (IFileOriginEntryViewModel)new FileOriginEntryViewModel(
-                        _conn,
-                        _archiveInstaller,
+                .Transform(fileOrigin => (IFileOriginEntryViewModel)
+                    new FileOriginEntryViewModel(
+                        conn,
+                        archiveInstaller,
                         LoadoutId,
-                        fileOrigin
+                        fileOrigin,
+                        viewModCommand
                     )
                 )
                 .Bind(out _fileOrigins);
-
+        
+            entriesObservable.SubscribeWithErrorLogging().DisposeWith(d);
+        });
+        
         this.WhenActivated(d =>
         {
-            entriesObservable.SubscribeWithErrorLogging().DisposeWith(d);
+            var canAddMod = new Subject<bool>();
+            var canAddAdvancedMod = new Subject<bool>();
+            this.WhenAnyValue(view => view.SelectedModsObservable)
+                .Where(obs => obs != null!)
+                .Select(observable =>
+                {
+                    return observable
+                        .WhenValueChanged(x => x.IsModAddedToLoadout)
+                        .Select(_ => SelectedModsCollection.Count > 0)
+                        .SubscribeWithErrorLogging(hasSelection =>
+                            {
+                                // Add (Advanced) is always available.
+                                canAddAdvancedMod.OnNext(hasSelection);
+                                
+                                // Add is only available if no mod is already added.
+                                canAddMod.OnNext(hasSelection && SelectedModsCollection.All(x => !x.IsModAddedToLoadout));
+                            }
+                        );
+                })
+                .Subscribe()
+                .DisposeWith(d);
+            
+            AddMod = ReactiveCommand.CreateFromTask(async (token) => await DoAddModImpl(null, token), canAddMod);
+            AddModAdvanced = ReactiveCommand.CreateFromTask(async (token) =>
+            {
+                await DoAddModImpl(_provider.GetKeyedService<IModInstaller>("AdvancedManualInstaller"), token);
+            }, canAddAdvancedMod);
         });
```

Comment:
```csharp
        var canAddMod = new Subject<bool>();
        var canAddAdvancedMod = new Subject<bool>();

        AddMod = ReactiveCommand.CreateFromTask(async cancellationToken => await DoAddModImpl(null, cancellationToken), canAddMod);
        AddModAdvanced = ReactiveCommand.CreateFromTask(async cancellationToken =>
        {
            await DoAddModImpl(_provider.GetKeyedService<IModInstaller>("AdvancedManualInstaller"), cancellationToken);
        }, canAddAdvancedMod);

        this.WhenActivated(d =>
        {
            var serialDisposable = new SerialDisposable();
            serialDisposable.DisposeWith(d);

            this.WhenAnyValue(vm => vm.SelectedModsObservable)
                .Select(observable =>
                {
                    return observable
                        .WhenValueChanged(x => x.IsModAddedToLoadout)
                        .Select(_ => SelectedModsCollection.Count > 0)
                        .SubscribeWithErrorLogging(hasSelection =>
                        {
                            // Add (Advanced) is always available.
                            canAddAdvancedMod.OnNext(hasSelection);

                            // Add is only available if no mod is already added.
                            canAddMod.OnNext(hasSelection && SelectedModsCollection.All(x => !x.IsModAddedToLoadout));
                        });
                })
                .SubscribeWithErrorLogging(disposable => serialDisposable.Disposable = disposable)
                .DisposeWith(d);
        });
```

Reply:
Question: Is there a reason the inner query can't also just do `DisposeWith(d)`?

I'm not exactly sure of the purpose of the `SerialDisposable` here. I can tell that the intent is to dispose the inner with the outer, as both produce observable sequences, but I don't see what we gain from moving it through `SerialDisposable`, which I believe is just a redirection layer.

Since the outer `SubscribeWithErrorLogging` has the result of the inner `SubscribeWithErrorLogging` as a parameter, which then gets `DisposeWith(d)`'d anyway/.

Reply:
Whenever the property `SelectedModsObservable` changes, we create a new observable stream and subscribe to it. When `SelectedModsObservable` changes again, we again create a new observable stream and subscribe to it.

If you don't dispose of the previous subscription when `SelectedModsObservable` changes again, you now have a leaking subscription.

It doesn't matter if `SelectedModsObservable` will only change once, that implementation detail is lost in this context. The best practice is to always dispose of subscription, and that's what you should do.

Reply:
All this discussion for functionality that wasn't part of the requirements and that we need to remove anyways. 
You should always be able to reinstall the mod. It should not matter if you have a copy installed already or not.

Reply:
It shouldn't be a leaking subscription if you `DisposeWith` the inner, however it will remain alive until the view itself is deactivated. Unless there's some very special implementation detail I'm unaware of.

Either way, using SerialDisposable is desirable here because we can dispose of previous items immediately, rather than keeping them forever until the view deactivates. I did figure it out eventually.